### PR TITLE
Display a report section on uncaught dartdoc tool failure.

### PR DESCRIPTION
**Pull request by isoos**
#4750

### CODE REVIEW

File: app/lib/dartdoc/dartdoc_runner.dart [hunk 1]
```diff
@@ -190,7 +191,12 @@ class DartdocJobProcessor extends JobProcessor {
     // We know that dartdoc will fail on this package, no reason to run it.
     if (packageStatus.isLegacy) {
       _logger.info('Package is on legacy SDK: $job.');
-      await _storeScoreCard(job, _emptyReport());
+      await _storeScoreCard(
+          job,
+          _emptyReport(
+            title: 'Did not run dartdoc',
+            description: 'Package supports only old SDK.',
```
Comment (sigurdm) [inline app/lib/dartdoc/dartdoc_runner.dart:198]:
```suggestion
            description: 'This package is not compatible with the current analysis SDK version',
```

File: app/lib/dartdoc/dartdoc_runner.dart [hunk 2]
```diff
@@ -200,7 +206,13 @@ class DartdocJobProcessor extends JobProcessor {
     if (packageStatus.isObsolete) {
       _logger
           .info('Package is older than two years and has newer release: $job.');
-      await _storeScoreCard(job, _emptyReport());
+      await _storeScoreCard(
+          job,
+          _emptyReport(
+            title: 'Did not run dartdoc',
+            description:
+                'Package version is older than two years and has newer release.',
```
Comment (sigurdm) [inline app/lib/dartdoc/dartdoc_runner.dart:214]:
```suggestion
                'Package version is older than two years and has a newer release.',
```

File: app/lib/dartdoc/dartdoc_runner.dart [hunk 3]
```diff
@@ -216,7 +228,12 @@ class DartdocJobProcessor extends JobProcessor {
         // These package versions are worth investigating, but we don't need
         // alerts on them.
         _logger.warning('Missing dartdoc report when error was present: $job.');
-        await _storeScoreCard(job, _emptyReport());
+        await _storeScoreCard(
+            job,
+            _emptyReport(
+              title: 'Failed to run dartdoc',
+              description: 'The dartdoc process timed out.',
```
Comment (sigurdm) [inline app/lib/dartdoc/dartdoc_runner.dart:235]: Do we have a way to indicate the time it took? Should we?

File: app/lib/dartdoc/dartdoc_runner.dart [hunk 4]
```diff
@@ -656,10 +673,26 @@ String _mergeOutput(ProcessResult pr, {bool compress = false}) {
...
+    documentationSection: ReportSection(
+      id: ReportSectionId.documentation,
+      title: documentationSectionTitle,
+      grantedPoints: 0,
+      maxPoints: 10,
+      summary: renderSimpleSectionSummary(
+        title: title,
+        description: description,
+        grantedPoints: 0,
+        maxPoints: 10,
```
Comment (sigurdm) [inline app/lib/dartdoc/dartdoc_runner.dart:693]: Is maxPoints 10 the right thing here? Maybe it doesn't really matter...

**Review by sigurdm — APPROVED** [review]
LGTM

File: app/lib/dartdoc/dartdoc_runner.dart (hunk 3 above)
Comment (isoos) [reply app/lib/dartdoc/dartdoc_runner.dart:235]: This is a tricky one: we have dartdoc runs that neither compete, nor time out (the timeout doesn't kill the process). Once we use the proper kill signal to do it (in this new release), I hope this class of error is gone and we can remove this block.

**Review by sigurdm — APPROVED** [review]
LGTM

**Pull request merged**